### PR TITLE
Nanohtml

### LINKED ISSUE #33: Move appendChild and attribute logic into files here
Currently the same logic exists across `yo-yoify` and `bel`. It would be nice if it could be more shared and driven by this library.

It also means an element created with a specific version of `bel` may not be compatible with a specific version of `yo-yoify`. So `yo-yoify` should really `require('bel/appendChild')` and `require('bel/attrs')` from the local element it is transforming to ensure compatibility.

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Changes `bel` into `nanohtml`. The main goal here is to merge `bel`, `yoyoify` and `pelo` into a single module that does all of these things. This should make it easier to maintain, test and point people to the right modules. Thanks!

## Tasks
- [x] port bel to nanohtml
- [x] port benchmarks
- [x] port tests
- [x] port types
- [ ] release major & deprecate bel
- [ ] release major & deprecate yo-yoify

Fixes #33

### CODE REVIEW

**Comment by goto-bus-stop** [comment]
re the license change https://github.com/choojs/choo/pull/582#issuecomment-368150805

**Comment by yoshuawuyts** [comment]
Reverted to MIT

**Comment by yoshuawuyts** [comment]
Ported benchmarks. Depends on https://github.com/mafintosh/nanobench/pull/7 to be merged, but it shouldn't hold up merging this I reckon. Thanks!

File: README.md [hunk 1]
```diff
@@ -1,186 +1,102 @@
...
-  `
-}
+## Static optimizations
+Parsing HTML has significant overhead. Being able to parse HTML statically,
+ahead of time can speed up rendering by about 2x.
 
+### Browserify
+```sh
+$ browserify -t nanohtml index.js > bundle.js
 ```
```
Comment (oe) [inline README.md]: MAYBE add a code example here too

File: package.json [hunk 2]
```diff
@@ -1,43 +1,64 @@
 {
-  "name": "bel",
-  "version": "5.1.7",
-  "description": "A simple extension to native elements",
-  "main": "index.js",
-  "browser": "browser.js",
+  "name": "nanohtml",
+  "description": "HTML template strings renderer for Node & Browsers.",
```
Comment (oe) [inline package.json]: `HTML template string renderer for the browser & Node`

i think we should display preference towards the browser here

File: types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+declare module "bel" {
```
Comment (oe) [inline types/index.d.ts]: shouldn't this be changed?

**Comment by yoshuawuyts** [comment]
@oe updated with all feedback!

**Comment by olstenlarck** [comment]
Damn.. Is `bel` 6 same as this? Why not just created `nanohtml` and let `bel` as it was on latest v5? Now updating services may update to v6 - it won't be a problem cuz is drop in replacement, but will pull more dependencies... Another way is to ignore the v6 update. ;/ Really sad. Bel worked both sides anyway.

**Comment by goto-bus-stop** [comment]
I thiiink Bel 6 is the same as Bel 5 but with a deprecation notice on npm. Not 100% sure, am on mobile!


On 14 March 2018 18:47:54 CET, Charlike Mike Reagent <notifications@github.com> wrote:
>Damn.. Is `bel` 6 same as this? Why not just created `nanohtml` and let
>`bel` as it was on latest v5? Now updating services may update to v6 -
>it won't be a problem cuz is drop in replacement, but will pull more
>dependencies... Another way is to ignore the v6 update. ;/ Really sad.
>Bel worked both sides anyway.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/choojs/nanohtml/pull/109#issuecomment-373114167

-- 
Sent from mobile. Please excuse my brevity.

**Comment by olstenlarck** [comment]
That would be good if so.

**Comment by yoshuawuyts** [comment]
@olstenlarck yep, @goto-bus-stop is right - that's exactly what we did. Most so if people perform a fresh `bel` install, they'll be nudged to move to `nanohtml` instead. Non-intrusive deprecations :tada:

**Pull request merged**